compiler-cli/src/run.rs

```diff
@@ -120,9 +120,29 @@ fn setup(
 
     let built = crate::build::main(paths, options, manifest)?;
 
+    //Warn incase the module being run has been as internal
+    let warning_emitter = WarningEmitter::new(Rc::new(ConsoleWarningEmitter));
+
+    //Warn incase the module being run has been as internal
+    if built.is_internal(&module.clone().into()).is_ok() {
```

Comment:
This emits a warning even when the module is not internal!

That aside, I think the fact this `is_internal` method has to internally check to find the module when that work has already been done in `find_package_config_for_module`. It shouldn't ever be possible for this code to run if the module was already found to be missing. Instead `find_package_config_for_module` could return this information.

Reply:
Sorry, what do you mean about a new PR?

Reply:
I was quoting this
> That aside, I think the fact this is_internal method has to internally check to find the module when that work has already been done in find_package_config_for_module. It shouldn't ever be possible for this code to run if the module was already found to be missing. Instead find_package_config_for_module could return this information.

I meant like, should I put this PR on hold until the work for `find_package_config_for_module` is done? Or is it better to merge this current PR and make a new PR in the future having the changes as you had mentioned once the required work for find_package_config_for_module is done?

Reply:
I don't know what work you mean, sorry. The feedback in this PR is for this PR, so it'll need to be resolved before it can be merged. Does that help?

Reply:
hey, yeah that helps, I had misunderstood a bit, thank you :)